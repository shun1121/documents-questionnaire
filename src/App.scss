html {
  background-color: #d3d3d3;
}
body {
  margin: 0px;
}
.App {
  text-align: center;
}
h1 > span {
  background-color: white;
  padding: 20px;
}
h1 {
  padding-top: 50px;
  height: 200px;
  color: black;
  margin-top: 0px;
}
.header {
  width: 100%;
  background-color: #1ea8fd;
}
p {
  text-align: left;
  font-size: 17px;
}
.form {
  background-color: white;
  margin: -120px auto 0;
  width: 70%;
  padding: 20px;
  margin-bottom: 80px;
}
form {
  width: 80%;
  margin: 0 auto;
}
.buttonWrapper {
  margin: 30px 0px
}
.questions {
  padding-top: 15px;
  margin-top: 35px;
  margin-bottom: 30px;
}
// overlayで囲んだ内容の表示位置などを決める
#overlay {
  position:fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(0,0,0,0.5);

  display: flex;
  overflow-y: scroll;
  overflow: auto;
}
#overlay2 {
  position:fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(0,0,0,0.5);

  display: flex;
  overflow-y: scroll;
  overflow: auto;
}
// .questionTtl {
//   padding-left: 10px;
//   text-align: left;
// }
// .content {
//   justify-content: center;
//   width: 80%;
//   background-color: white;
//   margin: auto;
//   padding: 20px 10px;
// }
// .resultWrapper {
//   display: flex;
//   flex-wrap: wrap;
// }
// .question {
//   width: 51%;
//   padding-right: 3px;
//   text-align: right;
//   border-right: 1px solid black;
// }
// .answer {
//   width: 48%;
//   word-wrap: break-word;
//   min-width: 48%;
//   text-align: center;
// }
// .close {
//   margin: 30px 0px;
// }
// .firstColumn {
//   margin-top: 30px;
// }
// .lastColumn {
//   margin-top: 15px;
//   margin-bottom: 40px;
// }
.caution {
  margin-top: 30px;
}