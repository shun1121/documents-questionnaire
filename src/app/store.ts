import { configureStore } from "@reduxjs/toolkit"
import QuestionnaireReducer from "../features/questionnaire/questionnaireSlice"

export const store = configureStore({
  reducer: {
    questionnaire: QuestionnaireReducer,
  },
})

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch
