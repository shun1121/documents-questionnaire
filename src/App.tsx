import React from "react"
import Checkbox from "./components/checkbox"
import InputText from "./components/inputText"
import QuestionnaireTitle from "./components/questionnaireTitle"
import Questions from "./components/questions"
import SelectBox from "./components/selectBox"
import TableSelect from "./components/tableSelect"
import "./App.scss"
import TestCheckbox from "./components/TestCheckbox"

const App = () => {
  return (
    <div className="App">
      <QuestionnaireTitle />
      <Questions questionNum="1" />
      <TableSelect />
      <Questions questionNum="2" />
      <Checkbox />
      {/* <Checkbox text="読む" />
      <Checkbox text="書く" />
      <Checkbox text="話す" />
      <Checkbox text="聞く" /> */}
      <Questions questionNum="3" />
      <TestCheckbox />  
      <SelectBox time="時間" />
      <Questions questionNum="4" />
      <InputText />
    </div>
  )
}

export default App
