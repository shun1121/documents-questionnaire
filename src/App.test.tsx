import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';
import { Provider } from 'react-redux'
import { store } from './app/store'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <App />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});