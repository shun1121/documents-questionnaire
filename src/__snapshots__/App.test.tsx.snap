// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <header
    className="header"
  >
    <div>
      <h1>
        <span>
          日本の英語教育に関するアンケート
        </span>
      </h1>
    </div>
  </header>
  <div
    className="form"
  >
    <form
      onSubmit={[Function]}
    >
      <p
        className="questions"
        data-testid="q1"
      >
        1. 以下の質問にお答えください。
      </p>
      <div>
        <table>
          <thead>
            <tr>
              <th
                className="th1"
              >
                質問
              </th>
              <th
                className="th2"
              >
                とても
                <br />
                同意
              </th>
              <th
                className="th3"
              >
                少し
                <br />
                同意
              </th>
              <th
                className="th4"
              >
                あまり
                <br />
                同意しない
              </th>
              <th
                className="th5"
              >
                全く
                <br />
                同意しない
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              className="1"
            >
              <td
                className="questionTtl td1"
              >
                <div
                  className="Question"
                >
                  <p
                    className="QuestionNum"
                  >
                    1. 
                  </p>
                  <p>
                    英語の授業中、教師は英語のみで教えるべきだ。
                  </p>
                </div>
              </td>
              <td
                className="td2"
              >
                <input
                  data-testid="button1-1"
                  id="button1-1"
                  name="q1"
                  onChange={[Function]}
                  type="radio"
                  value="とても同意"
                />
                <label
                  className="customButton"
                  htmlFor="button1-1"
                />
              </td>
              <td
                className="td3"
              >
                <input
                  data-testid="button1-2"
                  id="button1-2"
                  name="q1"
                  onChange={[Function]}
                  type="radio"
                  value="少し同意"
                />
                <label
                  className="customButton"
                  htmlFor="button1-2"
                />
              </td>
              <td
                className="td4"
              >
                <input
                  data-testid="button1-3"
                  id="button1-3"
                  name="q1"
                  onChange={[Function]}
                  type="radio"
                  value="あまり同意しない"
                />
                <label
                  className="customButton"
                  htmlFor="button1-3"
                />
              </td>
              <td
                className="td5"
              >
                <input
                  data-testid="button1-4"
                  id="button1-4"
                  name="q1"
                  onChange={[Function]}
                  type="radio"
                  value="全く同意しない"
                />
                <label
                  className="customButton"
                  htmlFor="button1-4"
                />
              </td>
            </tr>
            <tr
              className="2"
            >
              <td
                className="questionTtl td1"
              >
                <div
                  className="Question"
                >
                  <p
                    className="QuestionNum"
                  >
                    2. 
                  </p>
                  <p>
                    英語の例文や文章の音読をもっと取り入れるべきだ
                  </p>
                </div>
              </td>
              <td
                className="td2"
              >
                <input
                  id="button2-1"
                  name="q2"
                  onChange={[Function]}
                  type="radio"
                  value="とても同意"
                />
                <label
                  className="customButton"
                  htmlFor="button2-1"
                />
              </td>
              <td
                className="td3"
              >
                <input
                  id="button2-2"
                  name="q2"
                  onChange={[Function]}
                  type="radio"
                  value="少し同意"
                />
                <label
                  className="customButton"
                  htmlFor="button2-2"
                />
              </td>
              <td
                className="td4"
              >
                <input
                  id="button2-3"
                  name="q2"
                  onChange={[Function]}
                  type="radio"
                  value="あまり同意しない"
                />
                <label
                  className="customButton"
                  htmlFor="button2-3"
                />
              </td>
              <td
                className="td5"
              >
                <input
                  id="button2-4"
                  name="q2"
                  onChange={[Function]}
                  type="radio"
                  value="全く同意しない"
                />
                <label
                  className="customButton"
                  htmlFor="button2-4"
                />
              </td>
            </tr>
            <tr
              className="3"
            >
              <td
                className="questionTtl td1"
              >
                <div
                  className="Question"
                >
                  <p
                    className="QuestionNum"
                  >
                    3. 
                  </p>
                  <p>
                    中学英語段階で基本の発音記号は教えておくべきだ。
                  </p>
                </div>
              </td>
              <td
                className="td2"
              >
                <input
                  id="button3-1"
                  name="q3"
                  onChange={[Function]}
                  type="radio"
                  value="とても同意"
                />
                <label
                  className="customButton"
                  htmlFor="button3-1"
                />
              </td>
              <td
                className="td3"
              >
                <input
                  id="button3-2"
                  name="q3"
                  onChange={[Function]}
                  type="radio"
                  value="少し同意"
                />
                <label
                  className="customButton"
                  htmlFor="button3-2"
                />
              </td>
              <td
                className="td4"
              >
                <input
                  id="button3-3"
                  name="q3"
                  onChange={[Function]}
                  type="radio"
                  value="あまり同意しない"
                />
                <label
                  className="customButton"
                  htmlFor="button3-3"
                />
              </td>
              <td
                className="td5"
              >
                <input
                  id="button3-4"
                  name="q3"
                  onChange={[Function]}
                  type="radio"
                  value="全く同意しない"
                />
                <label
                  className="customButton"
                  htmlFor="button3-4"
                />
              </td>
            </tr>
            <tr
              className="4"
            >
              <td
                className="questionTtl td1"
              >
                <div
                  className="Question"
                >
                  <p
                    className="QuestionNum"
                  >
                    4. 
                  </p>
                  <p>
                    英語の授業では、英語らしく発音しなければならない環境を作るべきだ。
                  </p>
                </div>
              </td>
              <td
                className="td2"
              >
                <input
                  id="button4-1"
                  name="q4"
                  onChange={[Function]}
                  type="radio"
                  value="とても同意"
                />
                <label
                  className="customButton"
                  htmlFor="button4-1"
                />
              </td>
              <td
                className="td3"
              >
                <input
                  id="button4-2"
                  name="q4"
                  onChange={[Function]}
                  type="radio"
                  value="少し同意"
                />
                <label
                  className="customButton"
                  htmlFor="button4-2"
                />
              </td>
              <td
                className="td4"
              >
                <input
                  id="button4-3"
                  name="q4"
                  onChange={[Function]}
                  type="radio"
                  value="あまり同意しない"
                />
                <label
                  className="customButton"
                  htmlFor="button4-3"
                />
              </td>
              <td
                className="td5"
              >
                <input
                  id="button4-4"
                  name="q4"
                  onChange={[Function]}
                  type="radio"
                  value="全く同意しない"
                />
                <label
                  className="customButton"
                  htmlFor="button4-4"
                />
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <p
        className="questions"
        data-testid="q2"
      >
        2. 英語4技能でもっと学びたかった分野はどれですか。(複数選択有り)
      </p>
      <div>
        <div
          style={
            Object {
              "display": "flex",
              "justifyContent": "center",
            }
          }
        >
          <div
            className="checkbox-wrap"
            style={
              Object {
                "color": "black",
                "cursor": "pointer",
                "fontSize": "15px",
              }
            }
          >
            <input
              checked={false}
              id="読む"
              name="checkbox"
              onChange={[Function]}
              type="checkbox"
              value="読む"
            />
            <label
              className="paddingT"
              htmlFor="読む"
              style={
                Object {
                  "fontSize": "15px",
                }
              }
            >
              読む
            </label>
          </div>
          <div
            className="checkbox-wrap"
            style={
              Object {
                "color": "black",
                "cursor": "pointer",
                "fontSize": "15px",
              }
            }
          >
            <input
              checked={false}
              id="書く"
              name="checkbox"
              onChange={[Function]}
              type="checkbox"
              value="書く"
            />
            <label
              className="paddingT"
              htmlFor="書く"
              style={
                Object {
                  "fontSize": "15px",
                }
              }
            >
              書く
            </label>
          </div>
          <div
            className="checkbox-wrap"
            style={
              Object {
                "color": "black",
                "cursor": "pointer",
                "fontSize": "15px",
              }
            }
          >
            <input
              checked={false}
              id="話す"
              name="checkbox"
              onChange={[Function]}
              type="checkbox"
              value="話す"
            />
            <label
              className="paddingT"
              htmlFor="話す"
              style={
                Object {
                  "fontSize": "15px",
                }
              }
            >
              話す
            </label>
          </div>
          <div
            className="checkbox-wrap"
            style={
              Object {
                "color": "black",
                "cursor": "pointer",
                "fontSize": "15px",
              }
            }
          >
            <input
              checked={false}
              id="聞く"
              name="checkbox"
              onChange={[Function]}
              type="checkbox"
              value="聞く"
            />
            <label
              className="paddingT"
              htmlFor="聞く"
              style={
                Object {
                  "fontSize": "15px",
                }
              }
            >
              聞く
            </label>
          </div>
        </div>
      </div>
      <p
        className="questions"
        data-testid="q3"
      >
        3. あなたは一週間に英語を何時間勉強していますか。
      </p>
      <div
        className="wrapper"
      >
        <div
          className="selectbox select"
        >
          <select
            name="time"
            onChange={[Function]}
          >
            <option
              hidden={true}
            >
              時間を選択してください。
            </option>
            <option
              value="0~2"
            >
              0~2
            </option>
            <option
              value="2~4"
            >
              2~4
            </option>
            <option
              value="4~6"
            >
              4~6
            </option>
            <option
              value="6~8"
            >
              6~8
            </option>
            <option
              value="8~"
            >
              8~
            </option>
          </select>
        </div>
        <p
          className="time"
        >
          時間
        </p>
      </div>
      <p
        className="questions"
        data-testid="q4"
      >
        4. 日本の英語教育に取り入れるべきだと思うものを記述してください(400字以内)
      </p>
      <textarea
        onChange={[Function]}
        value=""
      />
      <div
        className="buttonWrapper"
        id="buttonWrapper"
      >
        <button
          className="storybook-button button-primary"
          id="button"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": undefined,
              "border": "none",
              "color": undefined,
              "cursor": "pointer",
              "padding": "0.5rem 1rem",
            }
          }
        >
          回答状況を確認
        </button>
      </div>
    </form>
  </div>
</div>
`;
